applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c

```diff
@@ -313,6 +344,23 @@ static void client_init(void)
 	client.tx_buf = tx_buffer;
 	client.tx_buf_size = sizeof(tx_buffer);
 
+	/* MQTT Will configuration */
+	struct mqtt_topic will_topic;
```

Comment:
Nothing guarantees that these memory areas exist after you exit this function. These should be static or alternatively defined in previous level, and brought as parameters, if only used in mqtt_connect.

Reply:
Yes they should be static